### DISCUSSION TITLE
Xidlehook bash script to dim screen automatically cancels without mouse trigger

### DISCUSSION THREAD

**Discussion by leminhman0312** [i3/i3#5532] [Q&A]
Hi all. This is my bash script to dim the screen, according to battery status (AC or Battery):
```
#!/bin/bash

export PRIMARY_DISPLAY="$(xrandr | awk '/ primary/{print $1}')"
bat=$(cat /sys/class/power_supply/AC/online)

if [[ $bat -eq 1 ]]
then
    # for AC
    xidlehook --not-when-fullscreen --not-when-audio --timer 30 'xrandr --output "$PRIMARY_DISPLAY" --brightness .4' \
    'xrandr --output "$PRIMARY_DISPLAY" --brightness 1'
else
    # for BATTERY
    xidlehook --not-when-fullscreen --not-when-audio --timer 30 'xrandr --output "$PRIMARY_DISPLAY" --brightness .2' \
    'xrandr --output "$PRIMARY_DISPLAY" --brightness 1'
fi
```
The script dims well with the preset timer and brightness reduction.  However, it only dims for like 2 seconds and then automatically trigger the canceller part of xidlehook: `xrandr --output "$PRIMARY_DISPLAY" --brightness 1` without needing any mouse movements. 

Any help is greatly appreciated.

**Comment by orestisfl** [comment]
Format your code with ``` to make it display correctly

```
like this
```

![image](https://github.com/i3/i3/assets/5778622/575bf03e-dfdb-42c5-9e7e-a84c1f2f8e13)

**Comment by orestisfl** [comment]
This mostly depends on how `xidlehook` detects activity and shouldn't be i3-specific. But, to be sure, you can try it with a different window manager like `openbox` and see if the issue persists
